snowpack/src/commands/install.ts

```diff
@@ -226,13 +227,21 @@ export async function install(
       dest: destLoc,
       externalPackage: externalPackages,
       sourceMap,
-      env,
+      env: userEnv,
       rollup: userDefinedRollup,
       treeshake: isTreeshake,
       polyfillNode,
     },
   } = config;
 
+  const env = {
```

Comment:
So we actually have some great code here that you could bring back from older version of Snowpack, where this was the main way that we did process.env replacement. For example, I'd love to see this function brought back (but with everything besides env removed: https://github.com/pikapkg/snowpack/blob/v2.6/src/commands/install.ts#L127

That way, your plugin usage is simplified to: 
```
      rollupPluginReplace(getRollupReplaceKeys(userEnv)),
```

Reply:
I don't think we want to quite do that.  `getRollupReplaceKeys` conflates computing the `env` dictionary and stringifying it for replacement.  In my approach, I compute `env` first so that both the replacement and the process polyfill can use it, guaranteeing that they'll have the same `process.env` values.

However, my code is almost exactly the same -- it's just split into two parts.  I changed it to use `JSON.stringify` instead, and I can break out the two `reduce`s into functions if you'd like but I think they're so small it won't really improve readability much.

Reply:
Okay, that makes sense. The `install()` function is large enough as it is, so In that case it would still be great to move this specific chunk of logic into a separate `const env = generateEnvObject(userEnv)` function.

Reply:
No problem, extracted.  But I work from email so I didn't see your edit about keeping the second reduce inline and extracted it as well -- hope it's OK.  😊